clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -3449,6 +3449,18 @@ def err_attribute_integers_only : Error<
 def warn_attribute_return_pointers_only : Warning<
   "%0 attribute only applies to return values that are pointers">,
   InGroup<IgnoredAttributes>;
+def warn_attribute_return_span_only
+    : Warning<"%0 attribute only applies to functions that return span-like "
+              "structures">,
+      InGroup<IgnoredAttributes>;
+def note_span_must_be_struct : Note<"span-like type must be a struct">;
```

Comment:
MUST it be a 'struct'? we probably can support class as well, right? 

Also, typically we phrase these in the reverse, something like: `returned type is not a struct/class type`